src/as-validator.c

```diff
@@ -1689,6 +1689,11 @@ as_validator_check_release (AsValidator *validator, xmlNode *node, AsFormatStyle
 		/* Neither timestamp, nor date property exists */
 		if (timestamp == NULL)
 			as_validator_add_issue (validator, node, "release-time-missing", "date");
+		else {
+			if (atoi(timestamp) <= 3000) {
```

Comment:
This arbitrarily checks for a "new enough" timestamp, it would be better to just check if the string value is a valid integer.
Like this:
```c
gint64 timestamp_int = 0;
errno = 0;
timestamp_int = g_ascii_strtoll (timestamp, NULL, 10);
if (errno != 0 && timestamp_int == 0)
    as_validator_add_issue (validator, node, "release-timestamp-invalid", timestamp);
```

Reply:
Would not be better to check both: Is it is a number and is it new enough? This would also catch cases where somebody try to use the Year as timestamp.

Reply:
On the one hand I don't like it that this would result in false-positives when some project decided to add a long release history going back decades. On the other hand, it is probably not a likely scenario at all.
But if we add a "Software too old" date, we should go all the way, I would likely accept something like this:
```c
/* check if the timestamp is both a number and a time after 1. Jan 2000 GMT+0 */
if (g_ascii_strtoll (timestamp, NULL, 10) < 946684800)
    as_validator_add_issue (validator, node, "release-timestamp-invalid", timestamp);
```

Reply:
What about using 3000 as above? It would filter out every Year and you can add Dates from the 90s.

Reply:
What if someone entered 20220520? ;-)
But okay, you convinced me! But please use `g_ascii_strtoll` (we don't want locale support and this is the much better choice here) and add a comment why 3000 is used to compare against :-)